src/Behat/Config/ConfigInterface.php

```diff
@@ -0,0 +1,10 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Behat\Config;
+
+interface ConfigInterface
```

Comment:
There are a lot of `ConfigInterface` out there already - I realise it's in a namespace, but something more explicit like `BehatConfigInterface` might help people avoid false positives when autocompleting in their IDE.

Reply:
I'm not sure about this comment, cause we may reuse this ConfigInterface for DTOs support.
like
```php
$config->withExtension(ConfigInterface $extensionConfig);
```

Reply:
Moreover, I think developers are now more careful about that. There are tons of cases like that.

Reply:
To be fair I don't think `BehatConfigInterface` prevents us using it for DTO / nested support - it's still "behat config", we're not saying it's e.g. `BehatConfigFileInterface` or `BehatRootConfigInterface`.

But I am not fixed on this.

I'm just in a mood because all day today my IDE has been forcing me to down-arrow past `Monolog\DateTimeImmutable` (we don't even use monolog, it's just pulled in as an "optional" dependency of another package) and every possible namespaced version of  `InvalidArgumentException` to get to the generic classes I want :frowning_face: 

I think developers are careful, it's just annoying. And like you say there are tons of cases, and really it's probably for the IDE to solve rather than us.